local/local.go

```diff
@@ -87,13 +102,98 @@ func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...Image
 		downloadBaseOnce: &sync.Once{},
 	}
 
-	for _, v := range ops {
-		image, err = v(image)
-		if err != nil {
-			return nil, err
-		}
+	image, err = processPreviousImageOption(image, imageOpts.prevImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = processBaseImageOption(image, imageOpts.baseImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = prepareNewWindowsImage(image)
+	if err != nil {
+		return image, err
+	}
+
+	return image, nil
+}
+
+func processPlatformOption(defaultPlatform imgutil.Platform, optionPlatform imgutil.Platform) (imgutil.Platform, error) {
+	if optionPlatform.OS != "" && optionPlatform.OS != defaultPlatform.OS {
+		return imgutil.Platform{}, fmt.Errorf(`invalid os: platform os "%s" must match the daemon os "%s"`, optionPlatform.OS, defaultPlatform.OS)
+	}
+
+	if (optionPlatform != imgutil.Platform{}) {
+		return optionPlatform, nil
 	}
 
+	return defaultPlatform, nil
+}
+
+func processPreviousImageOption(image *Image, prevImageRepoName string, platform imgutil.Platform, dockerClient client.CommonAPIClient) (*Image, error) {
+	if prevImageRepoName == "" {
+		return image, nil
+	}
+
+	if _, err := inspectOptionalImage(dockerClient, prevImageRepoName, platform); err != nil {
+		return image, err
+	}
+
+	prevImage, err := NewImage(prevImageRepoName, dockerClient, FromBaseImage(prevImageRepoName))
+	if err != nil {
+		return image, errors.Wrapf(err, "failed to get previous image '%s'", prevImageRepoName)
+	}
+
+	image.prevImage = prevImage
+
+	return image, nil
```

Comment:
Because `image` is a pointer we can set `image.prevImage` without returning the modified `image`. I think we should remove the first result parameter to make this clear. Without reading the implementation the function signature would seem to imply that the provided `image` has not been mutated.

Reply:
I changed the function signatures in local and remote